analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/Checks/InvocationCheck.cs

```diff
@@ -26,27 +29,49 @@ namespace SonarAnalyzer.SymbolicExecution.Roslyn.Checks
 {
     internal class InvocationCheck : SymbolicCheck
     {
+        private static ImmutableArray<KnownType> throwHelperAttributes = new[]
+        {
+            KnownType.System_Diagnostics_CodeAnalysis_DoesNotReturnAttribute,
+            KnownType.JetBrains_Annotations_TerminatesProgramAttribute,
+        }.ToImmutableArray();
+
         public override ProgramState[] PreProcess(SymbolicContext context) =>
             context.Operation.Instance.Kind == OperationKindEx.Invocation
             && PreProcess(context, IInvocationOperationWrapper.FromOperation(context.Operation.Instance)) is { } newState
                 ? newState
                 : base.PreProcess(context);
 
-        private static ProgramState[] PreProcess(SymbolicContext context, IInvocationOperationWrapper invocation) =>
-            invocation.TargetMethod.IsAny(KnownType.System_String, nameof(string.IsNullOrEmpty), nameof(string.IsNullOrWhiteSpace))
-                ? ProcessStringIsNullOrEmpty(context, invocation)
-                : null;
+        private static ProgramState[] PreProcess(SymbolicContext context, IInvocationOperationWrapper invocation)
+        {
+            if (invocation.TargetMethod.IsAny(KnownType.System_String, nameof(string.IsNullOrEmpty), nameof(string.IsNullOrWhiteSpace)))
+            {
+                return ProcessStringIsNullOrEmpty(context.State, invocation);
+            }
+            else if (IsThrowHelper(invocation))
+            {
+                return EmptyStates; // Stop processing
+            }
+            return null;
```

Comment:
Enforces every other branch to take decision and return
```suggestion
            else
            {
                return null;
            }               
```